### DISCUSSION TITLE
MOAR `Server error: the request cannot be completed because you have completed your <a href="/youtube/v3/getting-started#quota">quota</a>., reason: quotaExceeded` issues

### DISCUSSION THREAD

**Discussion by stdedos** [SkyTubeTeam/SkyTube#1088] [Q&A]
Hello there!

I am getting again "the same Server error messages" as these ones:

- https://github.com/SkyTubeTeam/SkyTube/issues/759
- https://github.com/SkyTubeTeam/SkyTube/issues/797
- https://github.com/SkyTubeTeam/SkyTube/issues/999#issuecomment-1000984786

I was wondering if it's "just me", how can I troubleshoot it, if there's a "central source" describing what to do with these issues.

I have tried both with the "NewPipe API" and without, and I have also tried generating my own API key with Youtube (using these instructions: https://developers.google.com/youtube/v3/getting-started), unsuccessfully.

It would be awesome if there were clear instructions written e.g. in a Wiki stored here on the repo, if the instructions above are not the correct ones.

Also, for the tin-foil-hatters out there, it'd be also very nice, if it were possible to provide us instructions for this segment of the page:
![image](https://user-images.githubusercontent.com/133706/175830232-83c7cc1d-53e1-428d-97ec-cd3d99cd71f4.png)

and specifically for these fields:
![image](https://user-images.githubusercontent.com/133706/175830242-20ed5711-9f6e-47ff-96f2-39f540f72859.png)

Provided ofc that they are not private/confidential data (that ofc you cannot "just" share like that)

**Comment by gzsombor** [comment]
SkyTube could get a lot of information, without using the official Youtube API, just by scraping the page, or calling REST endpoints, etc - Thanks to NewPipeExtractor. There are a couple of things, for which, there is no suitable substitution, so it must use the Youtube API. 
* Fetching the content of "Featured tab" - because it uses a custom query, to get all the popular videos, which are released in the last 24 hours (or something like that). It's a shame, that this tab was the default opening tab of the application, until the last release yesterday :)
* Fetching the list of the created playlist for a channel
So even though, you switch on the 'NewPipe' backend, you could still see the same error, if you are unlucky. However, in the reverse situation, when you switch off the NewPipe backend, a couple of things are still using NewPipe - I think, because querying a lot of thing could quickly drain your quota. And also it drained the quota attached to the couple of keys shipping by SkyTube as of now.
 For which feature, do you want to have a Youtube API key?

I think, if you just want to use your app locally (or with in your family), you don't need to restrict your key to a package, or to a SHA-1 certificate fingerprint.
However, you can restrict the key to only the "Youtube Data API v3" in the API restrictions

I'm not sure, sharing the SHA-1 fingerprints for the signing certificate is a secure thing, or not, or could it lead to a more widespread usage of the key.
